deploy: set namespace as kube-system

tam7t opened the pull request:
**What this PR does / why we need it**:

Resolved #375. The CSI driver needs permissions across all namespaces
for secrets syncing and access to the host filesystem. A non-cluster
admin user should not have the ability to update or modify the CSI
driver due to these high level of privileges.

This change would require users upgrading from previous verisons of the
driver to run the following to cleanup old resources:

kubectl delete DaemonSet csi-secrets-store -namespace default
kubectl delete DaemonSet csi-secrets-store-windows -namespace default
kubectl delete ServiceAccount secrets-store-csi-driver -namespace default

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #375

aramase reviewed (commented):
We would need to add a big warning note to let users know that the manifests will now deploy in `kube-system` and the necessary actions to take if they're upgrading using the manifests.

Also, we need to modify these:
- https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/config/rbac/role_binding.yaml#L12
- https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/config/rbac/serviceaccount.yaml#L5

tam7t commented:
Ah thanks for reminder about the `config` folder. I've updated and re-ran `make manifests`. I've also updated the readme to include _a_ warning and some instructions, but open to suggestions to make it clearer.

ritazh commented:
/lgtm

aramase commented on README.md:
````diff
@@ -143,6 +143,17 @@ NAME
+to clean up old resources:
+
+```bash
+kubectl delete DaemonSet csi-secrets-store -namespace=default
````
```suggestion
kubectl delete daemonset csi-secrets-store --namespace=default
```

`-namespace` needs to be `--namespace`

aramase commented on README.md:
````diff
@@ -143,6 +143,17 @@ NAME
+
+```bash
+kubectl delete DaemonSet csi-secrets-store -namespace=default
+kubectl delete DaemonSet csi-secrets-store-windows -namespace=default
````
```suggestion
kubectl delete daemonset csi-secrets-store-windows --namespace=default
```

aramase commented on README.md:
````diff
@@ -143,6 +143,17 @@ NAME
+```bash
+kubectl delete DaemonSet csi-secrets-store -namespace=default
+kubectl delete DaemonSet csi-secrets-store-windows -namespace=default
+kubectl delete ServiceAccount secrets-store-csi-driver -namespace=default
````
```suggestion
kubectl delete serviceaccount secrets-store-csi-driver --namespace=default
```

aramase reviewed (commented):
/lgtm
/approve
